api4/post.go

```diff
@@ -153,6 +156,9 @@ func getPostsForChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 		}
 
 		list, err = c.App.GetPostsAfterPost(c.Params.ChannelId, afterPost, c.Params.Page, c.Params.PerPage)
+		if len(list.Order) > 0 {
+			previousPostId = afterPost
```

Comment:
This only works if the page number is 0

Reply:
Not sure if I understand correctly but this also works for page other than 0 with valid `afterPost` parameter.  I have a test that covered that.